app/models/author.rb

```diff
@@ -1,5 +1,5 @@
 class Author < ApplicationRecord
     def discount() 10 end
-    has_many :books
+    has_many :books, dependent: :destroy
```

Comment:
This seemed to work well for me, but I'm not sure if this is the generally accepted way to do this.

If so, would it be best to also include this modification to the `published` line below?

Reply:
Since you're currently using sqlite (which doesn't support proper foreign keys) this is the best way to do it. If you're using a database that supports foreign keys (e.g. postgres) then setting up the foreign key to `on delete cascade` and leaving out this option is a better solution, as having the DB maintain data integrity will always be more reliable than having Rails do it.

Reply:
This _is_ the "Rails way" to do it, and since you're currently using sqlite (which doesn't support proper foreign keys) this is also the best way to do it. If you're using a database that supports foreign keys (e.g. postgres) then setting up the foreign key to `on delete cascade` and leaving out this option is a better solution, as having the DB maintain data integrity will always be more reliable than having Rails do it.

Reply:
Regarding adding it to the `published` relationship: that depends on what behaviour we want when a publisher is deleted. Should all their books _also_ be deleted? Or should they just no longer have a publisher (in which case you'd want to use `dependent: :nullify`. For the sake of this project let's go with the later behaviour (keep published books if a publisher is deleted).

Reply:
Excellent point re: sqlite vs postgres